src/features/users/user-route.ts

```diff
@@ -205,6 +208,32 @@ router.get('/status',
         next(httpResponse.Ok('fetched user status', response));
     }));
 
+    router.post('/adminUpdate',
+    authenticationMiddleware,
+    validate(adminUpdateValidation),
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    asyncHandler(async (req: RederlyExpressRequest<any, unknown, AdminUpdateRequest.body, AdminUpdateRequest.query>, res: Response, next: NextFunction) => {
+        if (_.isNil(req.session)) {
+            throw new Error(Constants.ErrorMessage.NIL_SESSION_MESSAGE);
+        }
+
+        const role = req.rederlyUserRole ?? req.rederlyUser?.roleId ?? Role.STUDENT;
+        if (role !== Role.SUPERADMIN) {
+            throw new ForbiddenError('You do not have access to update payment information.');
+        }
+
+        const result = await User.update({
+            paidUntil: req.body.paidUntil,
```

Comment:
These are optional, what if you send it undefined for either value (since they can be undefined)?

I think this should be wrapped in `_.omitBy(updates, _.isUndefined)`